local-pv-provisioner/README.md

```diff
@@ -0,0 +1,91 @@
+[![Docker Repository on Quay](https://quay.io/repository/cybozu/local-pv-provisioner/status "Docker Repository on Quay")](https://quay.io/repository/cybozu/local-pv-provisioner)
+
+local-pv-provisioner
+====================
+
+`local-pv-provisioner` is a custom controller that creates [local](https://kubernetes.io/docs/concepts/storage/volumes/#local) PersistentVolume(PV) resources from devices that match the specified conditions.
+
+* The PVs are linked to a node by `ownerReferences` setting.
+* The PVs will be removed along with the deletion of the node.
+
+## How to discover devices
+
+`local-pv-provisioner` searches for devices according to `--device-path` and `--device-name-filter` options.
+
+* `--device-path` option specifies the path to search the devices.
+* `--device-name-filter` option filter the device names using a regular expression.
+
+If you specifies the following condition, all devices under `/dev/disk/by-path/` will be selected.
+
+```console
+$ local-pv-provisioner --device-path="/dev/disk/by-path/" --device-name-filter=".*"
+```
+
+## What resources are created
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: local-node1-pci-0000-3b-00.0-ata-1
+  ownerReferences:
+    - apiVersion: v1
+      kind: Node
+      name: node1
+      uid: a958d5a1-2644-4e26-abb3-f810576b2f7f
+spec:
+  capacity:
+    storage: 100Gi
+  volumeMode: Block
+  accessModes:
+  - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Retain
+  storageClassName: local-storage
+  local:
+    path: /dev/disk/by-path/pci-0000:3b:00.0-ata-1
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: kubernetes.io/hostname
+          operator: In
+          values:
+          - node1
+```
+
+* `metadata.name` is decided according to the following rules.
+   * The name is a concatenation of `local`, node name, and device name with `-`.
+   * If it contains characters other than alphabets, numbers, `-` and `.`, it is replaced with `-`.
+* `spec.storageClassName` is automatically set a value `local-storage`.
+* `spec.capacity.storage` is decided from the max capacity of the device.
+
```

Comment:
Please add a section to explain how to bind the created PV with a PVC.

Reply:
I added a Usage section.